src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingWorker.java

```diff
@@ -282,16 +326,22 @@ public void onWorldTick(@NotNull final TickEvent.WorldTickEvent event)
     public void serializeToView(@NotNull final ByteBuf buf)
     {
         super.serializeToView(buf);
-
-        buf.writeInt(worker == null ? 0 : worker.getId());
+        buf.writeInt(workers.size());
+        for (final CitizenData data : workers)
+        {
+            buf.writeInt(data == null ? 0 : data.getId());
+        }
     }
 
     /**
      * AbstractBuildingWorker View for clients.
      */
     public static class View extends AbstractBuildingHut.View
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce this class from 52 to the maximum allowed 30 or externalize it in a public class. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)